packages/react-instantsearch-core/src/widgets/Index.tsx

```diff
@@ -83,7 +83,8 @@ class Index extends Component<InnerProps, State> {
         multiIndexContext: this.state.indexContext,
       },
       this.props,
-      undefined
+      undefined,
+      { displayName: 'AlgoliaIndex', getProvidedProps() {} }
```

Comment:
This dummy data reveals a flaw in the solution IMO. See my comment about using the Metadata API for this.

Reply:
yes, this doesn't really have a connector, could also make this key optional in the function

Reply:
what do you have as alternative in mind? I also thought instead of the widget requesting the rerender, it could be an argument to findResultsState (something like "renderPasses: 2")

Reply:
I like the approach of moving this to `findResultsState`, it feels less invasive and easier to control for the user.

Reply:
I like the approach of moving the logic to `findResultsState`. It feels less invasive and easier to control for users.